### PR TITLE
Refactor internal shapes code for point handling consistency

### PR DESCRIPTION
**Pull request by ejeschke**
Refactors shapes to have a more unified representation of points. 

This allows more efficient sharing of code that operates of points, removes "2D-isms" for a more unified approach to coordinate points in the API, and makes internal canvas code more consistent between shapes.
    
Objects with (x, y) and (x1, y1, x2, y2) attributes are refactored to store the points in an array of points notation similar to polygon-based shapes.  Properties are provided for backward-compatible access to these attributes.

### CODE REVIEW

**Comment by ejeschke** [comment]
@pllim, can you have a look/test when you get a chance?  Not urgent.  This is a follow on to #512.  As such, it shouldn't have any effect on users as it is mostly changes to "internal" APIs.

**Comment by ejeschke** [comment]
Rebased.

**Comment by ejeschke** [comment]
> can you have a look/test when you get a chance?

Please wait a little.  Pushing more commits.

**Comment by ejeschke** [comment]
Ok @pllim, sorry for the hold up.  I think this is ready for review/test.  I might push one more commit, but it would just be to update one of the example programs.  There are no special things to watch for--the updates are all in the path of regular rendering.

File: ginga/ImageView.py [hunk 1]
```diff
@@ -1545,25 +1547,21 @@ def _calc_bg_dimensions(self, scale_x, scale_y,
...
-        xlr, ylr = self.get_data_xy(win_wd, win_ht)
-        xll, yll = self.get_data_xy(0, win_ht)
+        pts_t = self.get_pan_rect().T
 
         # determine bounding box
-        a1 = min(xul, xur, xlr, xll)
-        b1 = min(yul, yur, ylr, yll)
-        a2 = max(xul, xur, xlr, xll)
-        b2 = max(yul, yur, ylr, yll)
+        a1, b1 = np.min(pts_t[0]), np.min(pts_t[1])
```
Comment (pllim) [inline ginga/ImageView.py]: I think you can just call `np.min()` one time and use `axis=1`. For example:
```python
>>> a = np.arange(10).reshape((2,5))
>>> a
array([[0, 1, 2, 3, 4],
       [5, 6, 7, 8, 9]])
>>> np.min(a, axis=1)
array([0, 5])
```

File: ginga/ImageView.py [hunk 2]
```diff
@@ -1545,25 +1547,21 @@ def _calc_bg_dimensions(self, scale_x, scale_y,
...
+        a2, b2 = np.max(pts_t[0]), np.max(pts_t[1])
```
Comment (pllim) [inline ginga/ImageView.py]: Same comment as `np.min()` above.

File: ginga/rv/plugins/Catalogs.py [hunk 3]
```diff
@@ -646,19 +646,19 @@ def filter_results(self, starlist, filter_obj):
         if filter_obj:
             num_cat = len(starlist)
             self.logger.debug("number of incoming stars=%d" % (num_cat))
-            # TODO: vectorize wcs lookup
-            coords = [image.radectopix(star['ra_deg'], star['dec_deg'])
-                      for star in starlist]
-            arr = numpy.array(coords)
-            self.logger.debug("arr.shape = %s" % str(arr.shape))
+            coords = numpy.asarray([(star['ra_deg'], star['dec_deg'])
```
Comment (pllim) [inline ginga/rv/plugins/Catalogs.py]: I see that you changed `numpy` to `np` in other files. Maybe change it here too for consistency.

Comment (ejeschke) [inline ginga/rv/plugins/Catalogs.py]: > are there any changes that are not backward compatible?

In terms of user-facing functions, only a few of the canvas API are affected, and these are preserved by compatibility functions at the bottom of `CanvasObject`.  If someone was hacking in their own transforms into the chain then yes, that would be affected.  But that is all fairly new code to ginga and still all fairly "internal".

File: ginga/ImageView.py (hunk 1 above)
Comment (ejeschke) [inline ginga/ImageView.py]: Good point. I'll refactor this a bit.

**Pull request merged**